README.md
# Color palettes for tailwindcss

## Installation

using `npm`
```bash
npm i @headwind-ui/color-palettes
```
using `yarn`
```bash
yarn add @headwind-ui/color-palettes
```
using `pnpm`
```bash
pnpm i @headwind-ui/color-palettes
```

## Usage

import a palette from `@headwind-ui/color-palettes`
```js
const { paletteOne } = require('@headwind-ui/color-palettes')

module.exports = {
  theme: {
   colors: {
     ...paletteOne,
   },
  },
}
```

import a color from `@headwind-ui/color-palettes`
```js
const { red } = require('@headwind-ui/color-palettes')

const color = red.hex // for get hex color
const color = red.hsl // for get hsl color
```

## Color Palettes

for view all color palettes, please visit [headwind-ui/color-palettes](headwind.urielcuriel.dev/color-palettes)



CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.0.4](https://github.com/UrielCuriel/color-palettes-taiwindcss/compare/v0.0.2...v1.0.4) (2022-08-14)

### [0.0.3](https://github.com/UrielCuriel/color-palettes-taiwindcss/compare/v0.0.2...v0.0.3) (2022-03-06)

### 0.0.2 (2022-03-06)


scripts/color-converter.js
const fs = require('fs')
const path = require('path')

/**
 * array colors to map of colors
 * @param {string[]} colors
 * @returns  {any}
 */
function arrayColorsToMap(colors) {
    const variants = ['100', '200', '300', '400', '500', '600', '700', '800', '900', '1000']
    const map = {}
    return variants.map((variant, i) => ({[variant]: colors[i]})).reduce((acc, curr) => ({...acc, ...curr}), map)
}
/**
 * sort hsl colors by brightness
 * @param {string[]} hslColors
 */
function sortHSLColorsByBrightness(hslColors) {
    return hslColors.sort((a, b) => {
        const aBrightness = getHSLBrightness(a)
        const bBrightness = getHSLBrightness(b)
        return bBrightness - aBrightness
    })
}

/**
 * sort hex colors by brightness
 * @param {*} hexColors
 * @returns {string[]}
 */
function sortHexColorsByBrightness(hexColors) {
    return hexColors.sort((a, b) => {
        const aBrightness = getHexBrightness(a)
        const bBrightness = getHexBrightness(b)
        return bBrightness - aBrightness
    })
}

/**
 * get brightness of color in HSL hsl(216, 100%, 93%)
 * @param {string} hslColor
 * @returns {number}
 */
function getHSLBrightness(hslColor) {
    const [, , brightness] = hslColor.split(',')
    return parseInt(brightness.replace('%', ''))
}

/**
 * get brightness of color in HEX #ff0000
 * @param {string} hexColor
 */
function getHexBrightness(hexColor) {
    //get rgb values
    const hasFullSpec = hexColor.length == 7
    const m = hexColor.substr(1).match(hasFullSpec ? /(\S{2})/g : /(\S{1})/g)
    if (!m) return 0
    const r = parseInt(m[0], 16)
    const g = parseInt(m[1], 16)
    const b = parseInt(m[2], 16)
    //get brightness
    return Math.round((r * 299 + g * 587 + b * 114) / 1000)
}

//read colors.txt
let colorsPath = path.join(__dirname, 'colors.txt')
let colors = fs.readFileSync(colorsPath, 'utf8')
let colorsArray = colors.split('\n')
let hslColors = []
let hexColors = []
let colorsMap = {}
colorsArray.forEach(color => {
    //continue if color is empty
    if (color.length === 0) return

    if (color.startsWith('#')) {
        hexColors.push(color)
    } else if (color.startsWith('hsl')) {
        hslColors.push(color)
    } else {
        //get color name remove spaces, parenthesis
        const colorName = color
            .replace(/\s+/g, '_')
            .replace(/\(|\)/g, '')
            .toLowerCase()
            .replace(/_([a-z])/g, match => match[1].toUpperCase())
        colorsMap[colorName] = {}
        colorsMap[colorName].hex = arrayColorsToMap(sortHexColorsByBrightness(hexColors))
        colorsMap[colorName].hsl = arrayColorsToMap(sortHSLColorsByBrightness(hslColors))
        hexColors = []
        hslColors = []
    }
})

//write src/colors.ts with every color exported
let colorsPathSrc = path.join(__dirname, '../src/colors.ts')
let colorsSrc = Object.entries(colorsMap).map(color => {
    return `export const ${color[0]} = {
    hex: ${JSON.stringify(color[1].hex)},
    hsl: ${JSON.stringify(color[1].hsl)}
}\n`
})
fs.writeFileSync(colorsPathSrc, colorsSrc.join('\n\n'), {flag: 'w'})


scripts/palette-generator.js
const fs = require('fs')
const path = require('path')

/**
 * @typedef Palette
 * @type {object}
 * @property {string[]} primary
 * @property {string[]} neutral
 * @property {string[]} support
 */

/**
 * get colors from palette
 * @param {Palette} palette
 */
function getColors(palette) {
    const {primary, neutral, support} = palette
    const colors = [...primary, ...neutral, ...support]
    return colors
}

/**
 * Generate palette structure from schema
 * @param {Palette} paletteDef
 * @param {'hex','hsl'} colorType
 */
function generatePalette(paletteDef, colorType) {
    const schema = JSON.parse(JSON.stringify(paletteDef))
    let palette = '{\n'
    Object.entries(schema).forEach(([key, value]) => {
        const defaultValue = value.splice(0, 1)
        palette += `${key}:{...${defaultValue}.${colorType},\n`
        if (value.length >= 1) {
            value.forEach(color => {
                palette += `'${color.replace(/([A-Z])/g, match => '-' + match[0].toLowerCase())}':${color}.${colorType},\n`
            })
        }
        palette += '},\n'
    })
    palette += '}'
    return palette
}

//read ./palettes.json
let palettesPath = path.join(__dirname, 'palettes.json')
let palettes = JSON.parse(fs.readFileSync(palettesPath, 'utf8'))
let importedColors = new Set()
let palettesSrc = ''
//get unique colors from palettes
Object.values(palettes).forEach(palette => {
    const colors = getColors(palette)
    colors.forEach(color => {
        importedColors.add(color)
    })
})
//sort colors by name
importedColors = Array.from(importedColors).sort()

//add colors to palettesSrc
palettesSrc += `import {${importedColors.join(', ')}} from './colors'\n\n`

//write src/palettes.ts with every palette exported
Object.entries(palettes).forEach(([paletteName, palette]) => {
    palettesSrc += `export const ${paletteName} = ${generatePalette(palette, 'hsl')}\n\n`
    palettesSrc += `export const ${paletteName}Hex = ${generatePalette(palette, 'hex')}\n\n`
})

fs.writeFileSync(path.join(__dirname, '../src/palettes.ts'), palettesSrc, {flag: 'w'})


src/colors.ts
export const red = {
    hex: {
        '100': '#FFEEEE',
        '200': '#FACDCD',
        '300': '#F29B9B',
        '400': '#E66A6A',
        '500': '#D64545',
        '600': '#BA2525',
        '700': '#A61B1B',
        '800': '#911111',
        '900': '#780A0A',
        '1000': '#610404'
    },
    hsl: {
        '100': 'hsl(360, 100%, 97%)',
        '200': 'hsl(360, 82%, 89%)',
        '300': 'hsl(360, 77%, 78%)',
        '400': 'hsl(360, 71%, 66%)',
        '500': 'hsl(360, 64%, 55%)',
        '600': 'hsl(360, 67%, 44%)',
        '700': 'hsl(360, 72%, 38%)',
        '800': 'hsl(360, 79%, 32%)',
        '900': 'hsl(360, 85%, 25%)',
        '1000': 'hsl(360, 92%, 20%)'
    }
}

export const redVivid = {
    hex: {
        '100': '#FFE3E3',
        '200': '#FFBDBD',
        '300': '#FF9B9B',
        '400': '#F86A6A',
        '500': '#EF4E4E',
        '600': '#E12D39',
        '700': '#CF1124',
        '800': '#AB091E',
        '900': '#8A041A',
        '1000': '#610316'
    },
    hsl: {
        '100': 'hsl(360, 100%, 95%)',
        '200': 'hsl(360, 100%, 87%)',
        '300': 'hsl(360, 100%, 80%)',
        '400': 'hsl(360, 91%, 69%)',
        '500': 'hsl(360, 83%, 62%)',
        '600': 'hsl(356, 75%, 53%)',
        '700': 'hsl(354, 85%, 44%)',
        '800': 'hsl(352, 90%, 35%)',
        '900': 'hsl(350, 94%, 28%)',
        '1000': 'hsl(348, 94%, 20%)'
    }
}

export const orange = {
    hex: {
        '100': '#FFEFE6',
        '200': '#FFD3BA',
        '300': '#FAB38B',
        '400': '#EF8E58',
        '500': '#E67635',
        '600': '#C65D21',
        '700': '#AB4E19',
        '800': '#8C3D10',
        '900': '#77340D',
        '1000': '#572508'
    },
    hsl: {
        '100': 'hsl(22, 100%, 95%)',
        '200': 'hsl(22, 100%, 86%)',
        '300': 'hsl(22, 92%, 76%)',
        '400': 'hsl(21, 83%, 64%)',
        '500': 'hsl(22, 78%, 55%)',
        '600': 'hsl(22, 71%, 45%)',
        '700': 'hsl(22, 74%, 38%)',
        '800': 'hsl(22, 79%, 31%)',
        '900': 'hsl(22, 80%, 26%)',
        '1000': 'hsl(22, 83%, 19%)'
    }
}

export const orangeVivid = {
    hex: {
        '100': '#FFE8D9',
        '200': '#FFD0B5',
        '300': '#FFB088',
        '400': '#FF9466',
        '500': '#F9703E',
        '600': '#F35627',
        '700': '#DE3A11',
        '800': '#C52707',
        '900': '#AD1D07',
        '1000': '#841003'
    },
    hsl: {
        '100': 'hsl(24, 100%, 93%)',
        '200': 'hsl(22, 100%, 85%)',
        '300': 'hsl(20, 100%, 77%)',
        '400': 'hsl(18, 100%, 70%)',
        '500': 'hsl(16, 94%, 61%)',
        '600': 'hsl(14, 89%, 55%)',
        '700': 'hsl(12, 86%, 47%)',
        '800': 'hsl(10, 93%, 40%)',
        '900': 'hsl(8, 92%, 35%)',
        '1000': 'hsl(6, 96%, 26%)'
    }
}

export const yellowVivid = {
    hex: {
        '100': '#FFFBEA',
        '200': '#FFF3C4',
        '300': '#FCE588',
        '400': '#FADB5F',
        '500': '#F7C948',
        '600': '#F0B429',
        '700': '#DE911D',
        '800': '#CB6E17',
        '900': '#B44D12',
        '1000': '#8D2B0B'
    },
    hsl: {
        '100': 'hsl(49, 100%, 96%)',
        '200': 'hsl(48, 100%, 88%)',
        '300': 'hsl(48, 95%, 76%)',
        '400': 'hsl(48, 94%, 68%)',
        '500': 'hsl(44, 92%, 63%)',
        '600': 'hsl(42, 87%, 55%)',
        '700': 'hsl(36, 77%, 49%)',
        '800': 'hsl(29, 80%, 44%)',
        '900': 'hsl(22, 82%, 39%)',
        '1000': 'hsl(15, 86%, 30%)'
    }
}

export const yellow = {
    hex: {
        '100': '#FFFAEB',
        '200': '#FCEFC7',
        '300': '#F8E3A3',
        '400': '#F9DA8B',
        '500': '#F7D070',
        '600': '#E9B949',
        '700': '#C99A2E',
        '800': '#A27C1A',
        '900': '#7C5E10',
        '1000': '#513C06'
    },
    hsl: {
        '100': 'hsl(45, 100%, 96%)',
        '200': 'hsl(45, 90%, 88%)',
        '300': 'hsl(45, 86%, 81%)',
        '400': 'hsl(43, 90%, 76%)',
        '500': 'hsl(43, 89%, 70%)',
        '600': 'hsl(42, 78%, 60%)',
        '700': 'hsl(42, 63%, 48%)',
        '800': 'hsl(43, 72%, 37%)',
        '900': 'hsl(43, 77%, 27%)',
        '1000': 'hsl(43, 86%, 17%)'
    }
}

export const limeGreen = {
    hex: {
        '100': '#F2FDE0',
        '200': '#E2F7C2',
        '300': '#C7EA8F',
        '400': '#ABDB5E',
        '500': '#94C843',
        '600': '#7BB026',
        '700': '#63921A',
        '800': '#507712',
        '900': '#42600C',
        '1000': '#2B4005'
    },
    hsl: {
        '100': 'hsl(83, 88%, 94%)',
        '200': 'hsl(84, 77%, 86%)',
        '300': 'hsl(83, 68%, 74%)',
        '400': 'hsl(83, 63%, 61%)',
        '500': 'hsl(83, 55%, 52%)',
        '600': 'hsl(83, 64%, 42%)',
        '700': 'hsl(83, 70%, 34%)',
        '800': 'hsl(83, 74%, 27%)',
        '900': 'hsl(81, 78%, 21%)',
        '1000': 'hsl(81, 86%, 14%)'
    }
}

export const limeGreenVivid = {
    hex: {
        '100': '#F8FFED',
        '200': '#E6FFBF',
        '300': '#CAFF84',
        '400': '#AFF75C',
        '500': '#8DED2D',
        '600': '#6CD410',
        '700': '#5CB70B',
        '800': '#399709',
        '900': '#2E7B06',
        '1000': '#1E5303'
    },
    hsl: {
        '100': 'hsl(83, 100%, 96%)',
        '200': 'hsl(83, 100%, 87%)',
        '300': 'hsl(86, 100%, 76%)',
        '400': 'hsl(88, 91%, 66%)',
        '500': 'hsl(90, 84%, 55%)',
        '600': 'hsl(92, 86%, 45%)',
        '700': 'hsl(92, 89%, 38%)',
        '800': 'hsl(100, 89%, 31%)',
        '900': 'hsl(99, 91%, 25%)',
        '1000': 'hsl(100, 93%, 17%)'
    }
}

export const green = {
    hex: {
        '100': '#E3F9E5',
        '200': '#C1EAC5',
        '300': '#A3D9A5',
        '400': '#7BC47F',
        '500': '#57AE5B',
        '600': '#3F9142',
        '700': '#2F8132',
        '800': '#207227',
        '900': '#0E5814',
        '1000': '#05400A'
    },
    hsl: {
        '100': 'hsl(125, 65%, 93%)',
        '200': 'hsl(126, 49%, 84%)',
        '300': 'hsl(122, 42%, 75%)',
        '400': 'hsl(123, 38%, 63%)',
        '500': 'hsl(123, 35%, 51%)',
        '600': 'hsl(122, 39%, 41%)',
        '700': 'hsl(122, 47%, 35%)',
        '800': 'hsl(125, 56%, 29%)',
        '900': 'hsl(125, 73%, 20%)',
        '1000': 'hsl(125, 86%, 14%)'
    }
}

export const greenVivid = {
    hex: {
        '100': '#E3F9E5',
        '200': '#C1F2C7',
        '300': '#91E697',
        '400': '#51CA58',
        '500': '#31B237',
        '600': '#18981D',
        '700': '#0F8613',
        '800': '#0E7817',
        '900': '#07600E',
        '1000': '#014807'
    },
    hsl: {
        '100': 'hsl(125, 65%, 93%)',
        '200': 'hsl(127, 65%, 85%)',
        '300': 'hsl(124, 63%, 74%)',
        '400': 'hsl(123, 53%, 55%)',
        '500': 'hsl(123, 57%, 45%)',
        '600': 'hsl(122, 73%, 35%)',
        '700': 'hsl(122, 80%, 29%)',
        '800': 'hsl(125, 79%, 26%)',
        '900': 'hsl(125, 86%, 20%)',
        '1000': 'hsl(125, 97%, 14%)'
    }
}

export const teal = {
    hex: {
        '100': '#EFFCF6',
        '200': '#C6F7E2',
        '300': '#8EEDC7',
        '400': '#65D6AD',
        '500': '#3EBD93',
        '600': '#27AB83',
        '700': '#199473',
        '800': '#147D64',
        '900': '#0C6B58',
        '1000': '#014D40'
    },
    hsl: {
        '100': 'hsl(152, 68%, 96%)',
        '200': 'hsl(154, 75%, 87%)',
        '300': 'hsl(156, 73%, 74%)',
        '400': 'hsl(158, 58%, 62%)',
        '500': 'hsl(160, 51%, 49%)',
        '600': 'hsl(162, 63%, 41%)',
        '700': 'hsl(164, 71%, 34%)',
        '800': 'hsl(166, 72%, 28%)',
        '900': 'hsl(168, 80%, 23%)',
        '1000': 'hsl(170, 97%, 15%)'
    }
}

export const tealVivid = {
    hex: {
        '100': '#F0FCF9',
        '200': '#C6F7E9',
        '300': '#8EEDD1',
        '400': '#5FE3C0',
        '500': '#2DCCA7',
        '600': '#17B897',
        '700': '#079A82',
        '800': '#048271',
        '900': '#016457',
        '1000': '#004440'
    },
    hsl: {
        '100': 'hsl(165, 67%, 96%)',
        '200': 'hsl(163, 75%, 87%)',
        '300': 'hsl(162, 73%, 74%)',
        '400': 'hsl(164, 70%, 63%)',
        '500': 'hsl(166, 64%, 49%)',
        '600': 'hsl(168, 78%, 41%)',
        '700': 'hsl(170, 91%, 32%)',
        '800': 'hsl(172, 94%, 26%)',
        '900': 'hsl(172, 98%, 20%)',
        '1000': 'hsl(176, 100%, 13%)'
    }
}

export const lightBlue = {
    hex: {
        '100': '#EBF8FF',
        '200': '#D1EEFC',
        '300': '#A7D8F0',
        '400': '#7CC1E4',
        '500': '#55AAD4',
        '600': '#3994C1',
        '700': '#2D83AE',
        '800': '#1D6F98',
        '900': '#166086',
        '1000': '#0B4F71'
    },
    hsl: {
        '100': 'hsl(201, 100%, 96%)',
        '200': 'hsl(200, 88%, 90%)',
        '300': 'hsl(200, 71%, 80%)',
        '400': 'hsl(200, 66%, 69%)',
        '500': 'hsl(200, 60%, 58%)',
        '600': 'hsl(200, 54%, 49%)',
        '700': 'hsl(200, 59%, 43%)',
        '800': 'hsl(200, 68%, 35%)',
        '900': 'hsl(200, 72%, 31%)',
        '1000': 'hsl(200, 82%, 24%)'
    }
}

export const lightBlueVivid = {
    hex: {
        '100': '#E3F8FF',
        '200': '#B3ECFF',
        '300': '#81DEFD',
        '400': '#5ED0FA',
        '500': '#40C3F7',
        '600': '#2BB0ED',
        '700': '#1992D4',
        '800': '#127FBF',
        '900': '#0B69A3',
        '1000': '#035388'
    },
    hsl: {
        '100': 'hsl(195, 100%, 95%)',
        '200': 'hsl(195, 100%, 85%)',
        '300': 'hsl(195, 97%, 75%)',
        '400': 'hsl(196, 94%, 67%)',
        '500': 'hsl(197, 92%, 61%)',
        '600': 'hsl(199, 84%, 55%)',
        '700': 'hsl(201, 79%, 46%)',
        '800': 'hsl(202, 83%, 41%)',
        '900': 'hsl(203, 87%, 34%)',
        '1000': 'hsl(204, 96%, 27%)'
    }
}

export const cyan = {
    hex: {
        '100': '#E0FCFF',
        '200': '#BEF8FD',
        '300': '#87EAF2',
        '400': '#54D1DB',
        '500': '#38BEC9',
        '600': '#2CB1BC',
        '700': '#14919B',
        '800': '#0E7C86',
        '900': '#0A6C74',
        '1000': '#044E54'
    },
    hsl: {
        '100': 'hsl(186, 100%, 94%)',
        '200': 'hsl(185, 94%, 87%)',
        '300': 'hsl(184, 80%, 74%)',
        '400': 'hsl(184, 65%, 59%)',
        '500': 'hsl(185, 57%, 50%)',
        '600': 'hsl(185, 62%, 45%)',
        '700': 'hsl(184, 77%, 34%)',
        '800': 'hsl(185, 81%, 29%)',
        '900': 'hsl(185, 84%, 25%)',
        '1000': 'hsl(184, 91%, 17%)'
    }
}

export const cyanVivid = {
    hex: {
        '100': '#E1FCF8',
        '200': '#C1FEF6',
        '300': '#92FDF2',
        '400': '#62F4EB',
        '500': '#3AE7E1',
        '600': '#1CD4D4',
        '700': '#0FB5BA',
        '800': '#099AA4',
        '900': '#07818F',
        '1000': '#05606E'
    },
    hsl: {
        '100': 'hsl(171, 82%, 94%)',
        '200': 'hsl(172, 97%, 88%)',
        '300': 'hsl(174, 96%, 78%)',
        '400': 'hsl(176, 87%, 67%)',
        '500': 'hsl(178, 78%, 57%)',
        '600': 'hsl(180, 77%, 47%)',
        '700': 'hsl(182, 85%, 39%)',
        '800': 'hsl(184, 90%, 34%)',
        '900': 'hsl(186, 91%, 29%)',
        '1000': 'hsl(188, 91%, 23%)'
    }
}

export const indigo = {
    hex: {
        '100': '#E0E8F9',
        '200': '#BED0F7',
        '300': '#98AEEB',
        '400': '#7B93DB',
        '500': '#647ACB',
        '600': '#4C63B6',
        '700': '#4055A8',
        '800': '#35469C',
        '900': '#2D3A8C',
        '1000': '#19216C'
    },
    hsl: {
        '100': 'hsl(221, 68%, 93%)',
        '200': 'hsl(221, 78%, 86%)',
        '300': 'hsl(224, 67%, 76%)',
        '400': 'hsl(225, 57%, 67%)',
        '500': 'hsl(227, 50%, 59%)',
        '600': 'hsl(227, 42%, 51%)',
        '700': 'hsl(228, 45%, 45%)',
        '800': 'hsl(230, 49%, 41%)',
        '900': 'hsl(232, 51%, 36%)',
        '1000': 'hsl(234, 62%, 26%)'
    }
}

export const blue = {
    hex: {
        '100': '#DCEEFB',
        '200': '#B6E0FE',
        '300': '#84C5F4',
        '400': '#62B0E8',
        '500': '#4098D7',
        '600': '#2680C2',
        '700': '#186FAF',
        '800': '#0F609B',
        '900': '#0A558C',
        '1000': '#003E6B'
    },
    hsl: {
        '100': 'hsl(205, 79%, 92%)',
        '200': 'hsl(205, 97%, 85%)',
        '300': 'hsl(205, 84%, 74%)',
        '400': 'hsl(205, 74%, 65%)',
        '500': 'hsl(205, 65%, 55%)',
        '600': 'hsl(205, 67%, 45%)',
        '700': 'hsl(205, 76%, 39%)',
        '800': 'hsl(205, 82%, 33%)',
        '900': 'hsl(205, 87%, 29%)',
        '1000': 'hsl(205, 100%, 21%)'
    }
}

export const blueVivid = {
    hex: {
        '100': '#E6F6FF',
        '200': '#BAE3FF',
        '300': '#7CC4FA',
        '400': '#47A3F3',
        '500': '#2186EB',
        '600': '#0967D2',
        '700': '#0552B5',
        '800': '#03449E',
        '900': '#01337D',
        '1000': '#002159'
    },
    hsl: {
        '100': 'hsl(202, 100%, 95%)',
        '200': 'hsl(204, 100%, 86%)',
        '300': 'hsl(206, 93%, 73%)',
        '400': 'hsl(208, 88%, 62%)',
        '500': 'hsl(210, 83%, 53%)',
        '600': 'hsl(212, 92%, 43%)',
        '700': 'hsl(214, 95%, 36%)',
        '800': 'hsl(215, 96%, 32%)',
        '900': 'hsl(216, 98%, 25%)',
        '1000': 'hsl(218, 100%, 17%)'
    }
}

export const indigoVivid = {
    hex: {
        '100': '#D9E8FF',
        '200': '#B0D0FF',
        '300': '#88B1FC',
        '400': '#5E8AEE',
        '500': '#3A66DB',
        '600': '#2251CC',
        '700': '#1D3DBF',
        '800': '#132DAD',
        '900': '#0B1D96',
        '1000': '#061178'
    },
    hsl: {
        '100': 'hsl(216, 100%, 93%)',
        '200': 'hsl(216, 100%, 85%)',
        '300': 'hsl(219, 95%, 76%)',
        '400': 'hsl(222, 81%, 65%)',
        '500': 'hsl(224, 69%, 54%)',
        '600': 'hsl(223, 71%, 47%)',
        '700': 'hsl(228, 74%, 43%)',
        '800': 'hsl(230, 80%, 38%)',
        '900': 'hsl(232, 86%, 32%)',
        '1000': 'hsl(234, 90%, 25%)'
    }
}

export const purple = {
    hex: {
        '100': '#EAE2F8',
        '200': '#CFBCF2',
        '300': '#A081D9',
        '400': '#8662C7',
        '500': '#724BB7',
        '600': '#653CAD',
        '700': '#51279B',
        '800': '#421987',
        '900': '#34126F',
        '1000': '#240754'
    },
    hsl: {
        '100': 'hsl(262, 61%, 93%)',
        '200': 'hsl(261, 68%, 84%)',
        '300': 'hsl(261, 54%, 68%)',
        '400': 'hsl(261, 47%, 58%)',
        '500': 'hsl(262, 43%, 51%)',
        '600': 'hsl(262, 48%, 46%)',
        '700': 'hsl(262, 60%, 38%)',
        '800': 'hsl(262, 69%, 31%)',
        '900': 'hsl(262, 72%, 25%)',
        '1000': 'hsl(263, 85%, 18%)'
    }
}

export const magenta = {
    hex: {
        '100': '#F5E1F7',
        '200': '#ECBDF2',
        '300': '#CE80D9',
        '400': '#BB61C7',
        '500': '#AD4BB8',
        '600': '#A23DAD',
        '700': '#90279C',
        '800': '#7C1A87',
        '900': '#671270',
        '1000': '#4E0754'
    },
    hsl: {
        '100': 'hsl(295, 58%, 93%)',
        '200': 'hsl(293, 67%, 85%)',
        '300': 'hsl(293, 54%, 68%)',
        '400': 'hsl(293, 48%, 58%)',
        '500': 'hsl(294, 43%, 51%)',
        '600': 'hsl(294, 48%, 46%)',
        '700': 'hsl(294, 60%, 38%)',
        '800': 'hsl(294, 68%, 32%)',
        '900': 'hsl(294, 72%, 25%)',
        '1000': 'hsl(295, 85%, 18%)'
    }
}

export const magentaVivid = {
    hex: {
        '100': '#FDEBFF',
        '200': '#F8C4FE',
        '300': '#F48FFF',
        '400': '#F368FC',
        '500': '#ED47ED',
        '600': '#E019D0',
        '700': '#CB10B8',
        '800': '#B30BA3',
        '900': '#960888',
        '1000': '#6E0560'
    },
    hsl: {
        '100': 'hsl(294, 100%, 96%)',
        '200': 'hsl(294, 97%, 88%)',
        '300': 'hsl(294, 100%, 78%)',
        '400': 'hsl(296, 96%, 70%)',
        '500': 'hsl(300, 82%, 60%)',
        '600': 'hsl(305, 80%, 49%)',
        '700': 'hsl(306, 85%, 43%)',
        '800': 'hsl(306, 88%, 37%)',
        '900': 'hsl(306, 90%, 31%)',
        '1000': 'hsl(308, 91%, 23%)'
    }
}

export const purpleVivid = {
    hex: {
        '100': '#F2EBFE',
        '200': '#DAC4FF',
        '300': '#B990FF',
        '400': '#A368FC',
        '500': '#9446ED',
        '600': '#8719E0',
        '700': '#7A0ECC',
        '800': '#690CB0',
        '900': '#580A94',
        '1000': '#44056E'
    },
    hsl: {
        '100': 'hsl(262, 90%, 96%)',
        '200': 'hsl(262, 100%, 88%)',
        '300': 'hsl(262, 100%, 78%)',
        '400': 'hsl(264, 96%, 70%)',
        '500': 'hsl(268, 82%, 60%)',
        '600': 'hsl(273, 80%, 49%)',
        '700': 'hsl(274, 87%, 43%)',
        '800': 'hsl(274, 87%, 37%)',
        '900': 'hsl(274, 87%, 31%)',
        '1000': 'hsl(276, 91%, 23%)'
    }
}

export const pink = {
    hex: {
        '100': '#FFE0F0',
        '200': '#FAB8D9',
        '300': '#F191C1',
        '400': '#E668A7',
        '500': '#DA4A91',
        '600': '#C42D78',
        '700': '#AD2167',
        '800': '#9B1B5A',
        '900': '#781244',
        '1000': '#5C0B33'
    },
    hsl: {
        '100': 'hsl(329, 100%, 94%)',
        '200': 'hsl(330, 87%, 85%)',
        '300': 'hsl(330, 77%, 76%)',
        '400': 'hsl(330, 72%, 65%)',
        '500': 'hsl(330, 66%, 57%)',
        '600': 'hsl(330, 63%, 47%)',
        '700': 'hsl(330, 68%, 40%)',
        '800': 'hsl(330, 70%, 36%)',
        '900': 'hsl(331, 74%, 27%)',
        '1000': 'hsl(330, 79%, 20%)'
    }
}

export const pinkVivid = {
    hex: {
        '100': '#FFE3EC',
        '200': '#FFB8D2',
        '300': '#FF8CBA',
        '400': '#F364A2',
        '500': '#E8368F',
        '600': '#DA127D',
        '700': '#BC0A6F',
        '800': '#A30664',
        '900': '#870557',
        '1000': '#620042'
    },
    hsl: {
        '100': 'hsl(341, 100%, 95%)',
        '200': 'hsl(338, 100%, 86%)',
        '300': 'hsl(336, 100%, 77%)',
        '400': 'hsl(334, 86%, 67%)',
        '500': 'hsl(330, 79%, 56%)',
        '600': 'hsl(328, 85%, 46%)',
        '700': 'hsl(326, 90%, 39%)',
        '800': 'hsl(324, 93%, 33%)',
        '900': 'hsl(322, 93%, 27%)',
        '1000': 'hsl(320, 100%, 19%)'
    }
}

export const blueGrey = {
    hex: {
        '100': '#F0F4F8',
        '200': '#D9E2EC',
        '300': '#BCCCDC',
        '400': '#9FB3C8',
        '500': '#829AB1',
        '600': '#627D98',
        '700': '#486581',
        '800': '#334E68',
        '900': '#243B53',
        '1000': '#102A43'
    },
    hsl: {
        '100': 'hsl(210, 36%, 96%)',
        '200': 'hsl(212, 33%, 89%)',
        '300': 'hsl(210, 31%, 80%)',
        '400': 'hsl(211, 27%, 70%)',
        '500': 'hsl(209, 23%, 60%)',
        '600': 'hsl(210, 22%, 49%)',
        '700': 'hsl(209, 28%, 39%)',
        '800': 'hsl(209, 34%, 30%)',
        '900': 'hsl(211, 39%, 23%)',
        '1000': 'hsl(209, 61%, 16%)'
    }
}

export const coolGrey = {
    hex: {
        '100': '#F5F7FA',
        '200': '#E4E7EB',
        '300': '#CBD2D9',
        '400': '#9AA5B1',
        '500': '#7B8794',
        '600': '#616E7C',
        '700': '#52606D',
        '800': '#3E4C59',
        '900': '#323F4B',
        '1000': '#1F2933'
    },
    hsl: {
        '100': 'hsl(216, 33%, 97%)',
        '200': 'hsl(214, 15%, 91%)',
        '300': 'hsl(210, 16%, 82%)',
        '400': 'hsl(211, 13%, 65%)',
        '500': 'hsl(211, 10%, 53%)',
        '600': 'hsl(211, 12%, 43%)',
        '700': 'hsl(209, 14%, 37%)',
        '800': 'hsl(209, 18%, 30%)',
        '900': 'hsl(209, 20%, 25%)',
        '1000': 'hsl(210, 24%, 16%)'
    }
}

export const grey = {
    hex: {
        '100': '#F7F7F7',
        '200': '#E1E1E1',
        '300': '#CFCFCF',
        '400': '#B1B1B1',
        '500': '#9E9E9E',
        '600': '#7E7E7E',
        '700': '#626262',
        '800': '#515151',
        '900': '#3B3B3B',
        '1000': '#222222'
    },
    hsl: {
        '100': 'hsl(0, 0%, 97%)',
        '200': 'hsl(0, 0%, 88%)',
        '300': 'hsl(0, 0%, 81%)',
        '400': 'hsl(0, 0%, 69%)',
        '500': 'hsl(0, 0%, 62%)',
        '600': 'hsl(0, 0%, 49%)',
        '700': 'hsl(0, 0%, 38%)',
        '800': 'hsl(0, 0%, 32%)',
        '900': 'hsl(0, 0%, 23%)',
        '1000': 'hsl(0, 0%, 13%)'
    }
}

export const warmGrey = {
    hex: {
        '100': '#FAF9F7',
        '200': '#E8E6E1',
        '300': '#D3CEC4',
        '400': '#B8B2A7',
        '500': '#A39E93',
        '600': '#857F72',
        '700': '#625D52',
        '800': '#504A40',
        '900': '#423D33',
        '1000': '#27241D'
    },
    hsl: {
        '100': 'hsl(40, 23%, 97%)',
        '200': 'hsl(43, 13%, 90%)',
        '300': 'hsl(40, 15%, 80%)',
        '400': 'hsl(39, 11%, 69%)',
        '500': 'hsl(41, 8%, 61%)',
        '600': 'hsl(41, 8%, 48%)',
        '700': 'hsl(41, 9%, 35%)',
        '800': 'hsl(37, 11%, 28%)',
        '900': 'hsl(40, 13%, 23%)',
        '1000': 'hsl(42, 15%, 13%)'
    }
}
